app/ante/ante.go

```diff
@@ -0,0 +1,53 @@
+package ante
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/crypto-org-chain/cronos/v2/x/cronos/keeper"
+	"github.com/crypto-org-chain/cronos/v2/x/cronos/types"
+	evmante "github.com/evmos/ethermint/app/ante"
+)
+
+// NewAnteHandler add additional logic on top of Ethermint's anteHandler
+func NewAnteHandler(options HandlerOptions) (sdk.AnteHandler, error) {
+	return func(
+		ctx sdk.Context, tx sdk.Tx, sim bool,
+	) (newCtx sdk.Context, err error) {
+		var anteHandler sdk.AnteHandler
+
+		defer evmante.Recover(ctx.Logger(), &err)
+
+		// Check msg authorization
+		for _, msg := range tx.GetMsgs() {
+			var permissionToCheck uint64
+			var accountToCheck sdk.AccAddress
+
+			switch v := msg.(type) {
```

Comment:
should this also have MsgUpdatePermissions ?

Reply:
I was thinking to exclude it because it does not check the permission but only that the sender is equal to admin but could also be added there with `permissionToCheck` = 0

Reply:
nvm `permissionToCheck = 0` won't work without some refactoring. Since the logic is a bit different I'll leave it out of the ante handler